File: ios/zendesk_sdk/Sources/zendesk_sdk/Models/ZendeskUser.swift
import Flutter

/// Immutable end-user identity passed to Zendesk (Value Object).
struct ZendeskUser {
    let name: String
    let emailId: String
    let userId: String
    let userType: String

    static let empty = ZendeskUser(name: "", emailId: "", userId: "", userType: "")

    var combinedName: String {
        "\(name) | UserID: \(userId)"
    }

    func baseTags(tripId: String? = nil) -> [String] {
        var tags = ["user_id:\(userId)", "mobile_app"]
        if !userType.isEmpty {
            tags.append("user_type:\(userType)")
        }
        if let tripId, !tripId.isEmpty {
            tags.append("trip_id:\(tripId)")
        }
        return tags
    }

    static func from(call: FlutterMethodCall) -> ZendeskUser {
        let args = call.arguments as? [String: Any]
        return ZendeskUser(
            name: args?[ZendeskSdkChannel.Argument.name] as? String ?? "",
            emailId: args?[ZendeskSdkChannel.Argument.emailId] as? String ?? "",
            userId: args?[ZendeskSdkChannel.Argument.userId] as? String ?? "",
            userType: args?[ZendeskSdkChannel.Argument.userType] as? String ?? ""
        )
    }
}

File: android/src/main/kotlin/com/example/zendesk_sdk/models/ZendeskUser.kt
package com.example.zendesk_sdk.models

import com.example.zendesk_sdk.ZendeskSdkChannel
import io.flutter.plugin.common.MethodCall

/// Immutable end-user identity passed to Zendesk (Value Object).
data class ZendeskUser(
    val name: String,
    val emailId: String,
    val userId: String,
    val userType: String,
) {
    val combinedName: String
        get() = "$name | UserID: $userId"

    fun baseTags(tripId: String? = null): List<String> {
        val tags = mutableListOf("user_id:$userId", "mobile_app")
        if (userType.isNotBlank()) {
            tags.add("user_type:$userType")
        }
        if (!tripId.isNullOrBlank()) {
            tags.add("trip_id:$tripId")
        }
        return tags
    }

    companion object {
        val EMPTY = ZendeskUser(name = "", emailId = "", userId = "", userType = "")

        fun fromMethodCall(call: MethodCall): ZendeskUser {
            return ZendeskUser(
                name = call.argument<String>(ZendeskSdkChannel.Argument.NAME) ?: "",
                emailId = call.argument<String>(ZendeskSdkChannel.Argument.EMAIL_ID) ?: "",
                userId = call.argument<String>(ZendeskSdkChannel.Argument.USER_ID) ?: "",
                userType = call.argument<String>(ZendeskSdkChannel.Argument.USER_TYPE) ?: "",
            )
        }
    }
}
